packages/app/src/cli/services/app/config/link.ts

```diff
@@ -68,13 +68,20 @@ async function loadRemoteApp(localApp: AppInterface, apiKey: string | undefined)
   return app
 }
 
+interface NewAppConfig {
```

Comment:
I don't love the updated typing of `mergeAppConfiguration` and am happy to fix forward on it. It's kinda weird because we're not really sure if it's valid app config when we're performing the merge (bc we might be creating something new, which won't have the mandatory fields), but want to be sure in the return.